### DISCUSSION TITLE
How to do if else in query

### DISCUSSION THREAD

**Discussion by kaisiant** [adonisjs/core#1715] [Help] [answered]
```
  public async index({ request }) {
    const data = request.only(['page', 'limit', 'code', 'displayName'])

    const page = data.page || 1
    const limit = data.limit || 10
    const categories = await Category.query()
      .where('code', 'ILIKE', `%${data.code}%`)
      .andWhere('displayName', 'ILIKE', `%${data.displayName}%`)
      .paginate(page, limit)
    return categories
  }
```
I only want to do `where code filter` when the code is provided else just do the displayName filter. Any idea?

**Comment by aayush123** [comment, marked as answer]
You can use conditional block to build query and then await it once you're done.

eg:
```javascript
const query = Category.query();
if (cond1) {
  query.where('field1', 'value1');
}
query.where('field2', 'value2');
const results = await query;
```

**Reply by kaisiant** [reply]
> You can use conditional block to build query and then await it once you're done.
> 
> eg:
> 
> ```js
> const query = Category.query();
> if (cond1) {
>   query.where('field1', 'value1');
> }
> query.where('field2', 'value2');
> const results = await query;
> ```

Cool. Thanks for the quick respond
